File: Algorithms/Strings/Two Characters.py
import os

#
# Complete the 'alternate' function below.
#
# The function is expected to return an INTEGER.
# The function accepts STRING s as parameter.
#

def alternate(s):
    # Write your code here
    unique_chars = set(s)
    max_len = 0

    for char1 in unique_chars:
        for char2 in unique_chars:
            if char1 != char2:
                filtered_str = [c for c in s if c == char1 or c == char2]
                if all(
                    filtered_str[i] != filtered_str[i + 1]
                    for i in range(len(filtered_str) - 1)
                ):
                    max_len = max(max_len, len(filtered_str))

    return max_len


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    l = int(input().strip())
    s = input()

    result = alternate(s)

    fptr.write(str(result) + '\n')
    fptr.close()

File: Algorithms/Strings/Two Characters.php
<?php

/*
 * Complete the 'alternate' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts STRING s as parameter.
 */

function alternate($s) {
    # Write your code here
    $max_length = 0;

    $unique_chars = array_unique(str_split($s));

    foreach($unique_chars as $char1) {
        foreach($unique_chars as $char2) {
            if($char1 != $char2) {
                $filtered_str = preg_replace("/[^$char1$char2]/", "", $s);
                if(ctype_alternate($filtered_str)) {
                    $max_length = max($max_length, strlen($filtered_str));
                }
            }
        }
    }

    return $max_length;
}

function ctype_alternate($s) {
    for($i = 0; $i < strlen($s) - 1; $i++) {
        if($s[$i] == $s[$i + 1]) {
            return False;
        }
    }
    return True;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$l = intval(trim(fgets(STDIN)));
$s = rtrim(fgets(STDIN), "\r\n");

$result = alternate($s);

fwrite($fptr, $result . "\n");
fclose($fptr);
